### PR TITLE
css fixes, R to refresh random, esc to close modal

### PR DESCRIPTION
**Pull request by Adjective-Object**
- fix issue with pages always being larger than the window
- add "press r to re-roll" on random quote page
- add "press esc to close" on quote modal

### CODE REVIEW

**Comment by Speculative** [comment]
Any chance you can bind that updateRandQuote so that when you click "Random" in the nav when you're already on /rand you can get a new random quote? Dunno what the easiest way to achieve that is without passing a callback around, maybe force a remount? That sounds like a weird solution. I dunno.

**Comment by nolanlum** [comment]
what if we make /rand redirect to the relevant quote's permalink route

**Comment by Adjective-Object** [comment]
> what if we make /rand redirect to the relevant quote's permalink route
seems like better overall UX. On rand we can get the URL of the quote and redirect to it that way, we just won't be able to use a normal `<Link>` element

Either that or we could just have `/rand` redirect to the appropriate permalinkquote? :thinking: 

In that case, the press R to reroll behavior should be moved to permalinkquote

**Comment by Speculative** [comment]
>In that case, the press R to reroll behavior should be moved to permalinkquote

Wait that don't exactly make sense to me. I'm viewing a quote that I clicked on from the main view and then I press R and get a random one? Do we gate it based on whether you got there from /rand?

**Comment by Adjective-Object** [comment]
> Do we gate it based on whether you got there from /rand?

this seems like unnecessary overkill to me

if we're already going to be implementing hjkl webcomic style keyboard controls as per gabe's suggestion, its not a stretch to add this to permalinkquote.

**Comment by Speculative** [comment]
Seems reasonable.

File: src/components/keybindings.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Header from "../containers/header";
+import SearchBar from "../components/searchbar";
+import PasteInput from "../components/pasteinput";
+
+import { Quote as ClassifiedQuote } from "../lib/classifyquote";
+import normalizeQuote from "../lib/normalizequote";
+import * as api from "../api/api";
+
+/**
+Component for managing keybdindings on a page.
```
Comment (nolanlum) [inline src/components/keybindings.tsx]: >keybdindings

File: src/components/keybindings.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+**/
+class Keybindings extends Component<{}, {}> {
+
+    private keyListener : EventListener;
+
+    constructor(props) {
+        super(props);
+        this.keyListener = this. __onKeyPress.bind(this);
```
Comment (nolanlum) [inline src/components/keybindings.tsx]: space after the period?

Comment (Adjective-Object) [inline src/components/keybindings.tsx]: I don't know _why_ this worked or passed tslint

Comment (Speculative) [inline src/components/keybindings.tsx]: Javascript lets you have whitespace after a `.` Dunno about tslint though 😛

**Pull request merged**